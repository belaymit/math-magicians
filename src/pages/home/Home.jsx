import React from 'react';
import './home.scss';

const Home = () => (
  <section className="home-container">
    <div className="container">
      <h2 className="header-title">Welcome to our page!</h2>
      <p className="text-desc">
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Magni autem mollitia ipsam nam consectetur voluptates dolorem
        dolorum magnam nobis, ad hic vel doloremque maiores possimus.
        Atque, voluptatum sequi. Voluptate eveniet eaque accusamus,
        repudiandae ratione magnam consequuntur pariatur voluptatum.
        Iure mollitia nostrum atque! Eaque amet sed similique, quisquam
        consequatur a, vero exercitationem reprehenderit autem mollitia
        blanditiis tempora. Reiciendis veniam quaerat fugiat nihil rem ut
        aliquid eos dolor vitae esse explicabo maiores saepe eius, assumenda
        incidunt excepturi iste.
      </p>
      <p className="text-desc">
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Magni autem mollitia ipsam nam consectetur voluptates dolorem
        dolorum magnam nobis, ad hic vel doloremque maiores possimus.
        Atque, voluptatum sequi. Voluptate eveniet eaque accusamus,
        repudiandae ratione magnam consequuntur pariatur voluptatum.
        Iure mollitia nostrum atque! Eaque amet sed similique, quisquam
        consequatur a, vero exercitationem reprehenderit autem mollitia
        blanditiis tempora. Reiciendis veniam quaerat fugiat nihil rem ut
        aliquid eos dolor vitae esse explicabo maiores saepe eius, assumenda
        incidunt excepturi iste.
      </p>
    </div>
  </section>
);

export default Home;
