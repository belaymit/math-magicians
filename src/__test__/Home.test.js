import { render, screen } from '@testing-library/react';
import Home from '../pages/home/Home';

require('@testing-library/jest-dom');

describe('This is a test for home component', () => {
  it('renders the Home component without any error', () => {
    const output = render(<Home />);
    expect(output).toMatchSnapshot();
  });

  it('The text "Welcome to our page!" should be rendered in Home component without any error', () => {
    render(<Home />);
    expect(screen.getByText('Welcome to our page!')).toBeInTheDocument();
  });
});
