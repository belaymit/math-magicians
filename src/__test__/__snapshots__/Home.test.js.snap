// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is a test for home component renders the Home component without any error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="home-container"
      >
        <div
          class="container"
        >
          <h2
            class="header-title"
          >
            Welcome to our page!
          </h2>
          <p
            class="text-desc"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Magni autem mollitia ipsam nam consectetur voluptates dolorem dolorum magnam nobis, ad hic vel doloremque maiores possimus. Atque, voluptatum sequi. Voluptate eveniet eaque accusamus, repudiandae ratione magnam consequuntur pariatur voluptatum. Iure mollitia nostrum atque! Eaque amet sed similique, quisquam consequatur a, vero exercitationem reprehenderit autem mollitia blanditiis tempora. Reiciendis veniam quaerat fugiat nihil rem ut aliquid eos dolor vitae esse explicabo maiores saepe eius, assumenda incidunt excepturi iste.
          </p>
          <p
            class="text-desc"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Magni autem mollitia ipsam nam consectetur voluptates dolorem dolorum magnam nobis, ad hic vel doloremque maiores possimus. Atque, voluptatum sequi. Voluptate eveniet eaque accusamus, repudiandae ratione magnam consequuntur pariatur voluptatum. Iure mollitia nostrum atque! Eaque amet sed similique, quisquam consequatur a, vero exercitationem reprehenderit autem mollitia blanditiis tempora. Reiciendis veniam quaerat fugiat nihil rem ut aliquid eos dolor vitae esse explicabo maiores saepe eius, assumenda incidunt excepturi iste.
          </p>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="home-container"
    >
      <div
        class="container"
      >
        <h2
          class="header-title"
        >
          Welcome to our page!
        </h2>
        <p
          class="text-desc"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Magni autem mollitia ipsam nam consectetur voluptates dolorem dolorum magnam nobis, ad hic vel doloremque maiores possimus. Atque, voluptatum sequi. Voluptate eveniet eaque accusamus, repudiandae ratione magnam consequuntur pariatur voluptatum. Iure mollitia nostrum atque! Eaque amet sed similique, quisquam consequatur a, vero exercitationem reprehenderit autem mollitia blanditiis tempora. Reiciendis veniam quaerat fugiat nihil rem ut aliquid eos dolor vitae esse explicabo maiores saepe eius, assumenda incidunt excepturi iste.
        </p>
        <p
          class="text-desc"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Magni autem mollitia ipsam nam consectetur voluptates dolorem dolorum magnam nobis, ad hic vel doloremque maiores possimus. Atque, voluptatum sequi. Voluptate eveniet eaque accusamus, repudiandae ratione magnam consequuntur pariatur voluptatum. Iure mollitia nostrum atque! Eaque amet sed similique, quisquam consequatur a, vero exercitationem reprehenderit autem mollitia blanditiis tempora. Reiciendis veniam quaerat fugiat nihil rem ut aliquid eos dolor vitae esse explicabo maiores saepe eius, assumenda incidunt excepturi iste.
        </p>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
